Shift3/boilerplate-client-react
# Chore(redux): Remove Redux and add analyze script

## Changes
1. Removed 'Redux' and all it's dependencies and types.
2. Added an 'Analyze' Script to work with Source-Map-Explorer.

## Purpose
1.) To remove unused libraries. We will be building with the Context API for now.
2.) Reduce bundle size and analyze our project, by adding an 'analyze' script to 'package.json'

## Approach
1.) Context API along with React hooks, will handle our state management.
2.) To reduce bundle size. We have added 'Source-Map-Explorer' as a dependency. The 'analyze' script will allow us to check our bundle and packages, and determine which dependencies or future dependencies are the heaviest, and which we prefer to use.

## Learning
For more information ('https://create-react-app.dev/docs/analyzing-the-bundle-size/'medium' ). 

## Testing
1. Pull in the changes to your local copy of this branch and run [`yarn build`]
2. Then run [`yarn analyze`]
3. Check the browser to view the analyzed data.

## Screenshots
![Screen Shot 2021-04-12 at 6 14 10 PM](https://user-images.githubusercontent.com/59614789/114469312-e3065f00-9bba-11eb-8570-1d0d3c7ba2f1.png)

### Closes #73
